File: my_vpn/app/src/main/java/com/myvpn/simple/TrojanConfig.java
package com.myvpn.simple;

import java.io.Serializable;

public class TrojanConfig implements Serializable {
    public String server;
    public int port;
    public String password;
    public String sni;
    public boolean allowInsecure = false;
    
    public TrojanConfig() {}
    
    public TrojanConfig(String server, int port, String password) {
        this.server = server;
        this.port = port;
        this.password = password;
    }
    
    public boolean isValid() {
        return server != null && !server.isEmpty() && 
               port > 0 && port <= 65535 && 
               password != null && !password.isEmpty();
    }
    
    @Override
    public String toString() {
        return "trojan://" + password + "@" + server + ":" + port;
    }
}

File: kt/app/src/main/kotlin/com/myvpn/simple/TrojanConfig.kt
package com.myvpn.simple

import java.io.Serializable

data class TrojanConfig(
    var server: String = "",
    var port: Int = 0,
    var password: String = "",
    var sni: String = "",
    var allowInsecure: Boolean = false,
    var remark: String = "",
    var country: String = ""
) : Serializable {

    fun isValid(): Boolean {
        return server.isNotEmpty() && port in 1..65535 && password.isNotEmpty()
    }

    override fun toString(): String {
        return "trojan://$password@$server:$port"
    }
}
